repository/src/main/java/org/apache/atlas/repository/store/graph/v2/preprocessor/ConnectionPreProcessor.java

```diff
@@ -171,13 +172,13 @@ private void processUpdateConnection(EntityMutationContext context,
             entity.setAttribute(QUALIFIED_NAME, vertexQName);
 
             RoleRepresentation representation = getKeycloakClient().getRoleByName(roleName);
-            String creatorUser = vertex.getProperty(CREATED_BY_KEY, String.class);
+           // String creatorUser = vertex.getProperty(CREATED_BY_KEY, String.class);
 
             if (connection.hasAttribute(ATTR_ADMIN_USERS)) {
                 List<String> newAdminUsers = (List<String>) connection.getAttribute(ATTR_ADMIN_USERS);
                 List<String> currentAdminUsers = (List<String>) existingConnEntity.getAttribute(ATTR_ADMIN_USERS);
-                if (StringUtils.isNotEmpty(creatorUser) && !newAdminUsers.contains(creatorUser)) {
-                    newAdminUsers.add(creatorUser);
+                if (CollectionUtils.isEmpty(newAdminUsers)) {
```

Comment:
Only checking for adminUsers attribute, ideally, we should check all 3 adminUsers, adminGoups & adminRoles are not empty
Imagine a scenario like no direct user is admin but All Admins is selected, so logically admins list is not empty but since we only check for adminUsers we it will throw this exception in such scenario

Reply:
I hope we do not want to change the behavior for the Connection creations, otherwise, the same check would have been applicable for the new Connection creation case